src/ngraph/op/experimental/batch_dot.cpp

```diff
@@ -0,0 +1,139 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "batch_dot.hpp"
+#include "ngraph/dimension.hpp"
+#include "ngraph/log.hpp"
+#include "ngraph/op/experimental/dyn_reshape.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::BatchDot::BatchDot(const shared_ptr<Node>& arg0,
```

Comment:
You know I'm not sure it's worth bikeshedding, but I wonder if `BatchMatMul` is a better name :)

Reply:
I see a good point here since we always deal with matrices here. I'll rename it.

Reply:
@jbobba If I rename this to `BatchMatMul`, I think the backend should be renamed too? What do you think? Keep it as `BatchDot` or rename?

Reply:
`BatchMatMul` seems to be the name frameworks use, so we should consistently go with that.

Reply:
Fixed.